soh/soh/Enhancements/mods.cpp

```diff
@@ -258,6 +258,28 @@ void RegisterRupeeDash() {
     });
 }
 
+void RegisterShadowTag() {
+    static uint16_t tagSpawnable = 0;
+    static uint16_t tagDelay = 60;
+
+    GameInteractor::Instance->RegisterGameHook<GameInteractor::OnPlayerUpdate>([]() {
+        if (!CVarGetInteger("gShadowTag", 0)) {
+            return;
+        }
+        if ((tagSpawnable == 0) && (tagDelay <= 0)) {
+            tagSpawnable = 1;
+            tagDelay = 0;
+            GameInteractor::RawAction::SpawnEnemyWithOffset(17, 3);
+        } else {
+            tagDelay--;
+        }
+    });
+    GameInteractor::Instance->RegisterGameHook<GameInteractor::OnSceneInit>([](int16_t sceneNum) { 
+        tagSpawnable = 0;
+        tagDelay = 60;
+    });
```

Comment:
Have you tried finding a different spot to place a hook so you don't need the timer stuff? For example, around the same spot where the game normally loads in the scene data and places enemies. If OnSceneInit is an accurate name, it also won't trigger when changing rooms inside dungeons, not sure if you tried that yet.